### PR TITLE
Filter separation

### PR DESCRIPTION
**Pull request by FWDekker**
The `ValueFilter` should be doing three things:
1. Filtering _out_ particular classes of `Value`s
2. Filtering _out_ `Value`s that use something from a user's own code (because we cannot use these statements)
3. Filtering _out_ `Value`s that do not use the library.

Previously, the `ValueFilter` did 1 and 2, but I was unable to implement 3 because it is incompatible with 2. This is because 2 is a universal check of absence ("for all not"), while 3 is an existential check of presence ("any").

Therefore, I have split the `ValueFilter` into three separate filters, each filter performing one of these three steps. I thought it would be easiest to do this using the `JimpleValueVisitor`, so I did.

### CODE REVIEW

File: modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt [hunk 1]
```diff
@@ -1,103 +1,163 @@
...
-import soot.jimple.UnopExpr
-import soot.jimple.internal.AbstractBinopExpr
+import soot.jimple.StaticFieldRef
+import soot.jimple.StaticInvokeExpr
 import soot.jimple.toolkits.infoflow.AbstractDataSource
 import soot.jimple.toolkits.thread.synchronization.NewStaticLock
-import soot.shimple.PhiExpr
-import soot.shimple.ShimpleExpr
+
+internal class ValueFilter(libraryProject: JavaProject) {
```
Comment (gandreadis) [inline modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Where did the Javadoc go? 😢

File: modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt [hunk 2]
```diff
@@ -1,103 +1,163 @@
...
+     */
+    override fun accumulate(result1: Boolean, result2: Boolean) = result1 && result2
+}
+
+/**
+ * Retains [Value]s that use the library somewhere.
+ *
+ * @property libraryProject a library project
+ */
+@Suppress("MethodOverloading", "TooManyFunctions") // That is how JimpleValueVisitor works
```
Comment (gandreadis) [inline modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: JimpleValueVisitor works by using too many functions? :P

Comment (FWDekker) [reply modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: Exactly!

File: modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt (hunk 1 above)
Comment (FWDekker) [reply modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: It moved to `ValueClassFilter` where I completely rewrote it :P

Comment (FWDekker) [reply modules/mining-pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/usagegraphgenerator/jimple/filters/ValueFilter.kt]: @gandreadis I've added the missing documentation

**Pull request merged**